### ISSUE TITLE
Conflicts with Mongoid search/index methods

### ISSUE THREAD

**Issue by christianhellsten** [karmi/retire#157]
This class:

```
class XXX
  index :slug

  # Must be included after Mongoid, Tire#index conflicts with Mongoid#index otherwise
  include Tire::Model::Search
  include Tire::Model::Callbacks
```

will not be searchable with Mongoid 2.3.4. The search method is implemented by Mongoid, so instead of performing a search through Tire it will return e.g.:

```
XXX.search('yyy')
=> [:ids,
 #<Mongoid::Criteria
  selector: {:_id=>"yyy", :deleted_at=>nil},
  options:  {},
  class:    XXX,
  embedded: false>
```

**Comment by karmi** [comment]
I'm not sure I understand the issue, could you be more specific please?

Recent releases of Tire do not include its methods in the class, when they are already defined. They are accessible via the `tire` proxy, eg. `MyModel.tire.search`.

**Comment by christianhellsten** [comment]
OK, that explains it. I was following the README instructions, which still recommend using Article.search, not Article.tire.search:

```
Article.search 'love'
```

I guess I should have read the part about Article.tire.search...

**Comment by karmi** [comment]
Yes, since `search` is mostly not used already... The README explains it in detail a bit further.

**Comment by christianhellsten** [comment]
Yes, thanks for the quick reply. Closing this issue, since it's well documented in the README.

**Comment by bashcoder** [comment]
For what it's worth, I did manage to dismiss the single mention of this in the README, probably because it is only said to be something you _could_ do, to be on the "safe side."

In my case, it was a _must_ do to in order get Tire working with Mongoid, otherwise I would get a "stack level too deep" error.  So I guess would suggest that the `Article.search` examples be changed to `Article.tire.search` instead, since that method would always work. If only for the Mongoid section.

**Comment by karmi** [comment]
@bashcoder It's really highly specific to _Mongoid_. You're right that in the _Mongoid_ part, we should use the `tire` proxy, good suggestion!

Update: Fixed in aa19638
